### DISCUSSION TITLE
Option to make clicking Today button open today's daily note

### DISCUSSION THREAD

**Discussion by JackALaing** [liamcain/obsidian-calendar-plugin#64] [Q&A]
I think clicking the Today button (between the arrows) should open today's Daily Note, but I recognize that not everyone will want that. Give us the option :)

**Comment by liamcain** [comment]
I'm always weary of giving users too much customization. I talked a bit about the design rationale for the "today" button here: https://github.com/liamcain/obsidian-calendar-plugin/issues/56#issuecomment-736525739

Is the main concern that the button's current behavior isn't intuitive? Or is it more so that you'd like a faster way to open today's daily note?

**Comment by JackALaing** [comment]
Coming from NotePlan, where the equivalent of clicking the Today button takes you to today's Daily Note, I often find myself instinctively clicking it then being disappointed.

At the moment the only function of the button is to bring you back to the current month if you're looking at a different month, which I don't do often. When I'm already in the current month, it does nothing. 

Perhaps there should be another layer to the logic of the button, where if you're already in the current month, rather than doing nothing, it opens the daily note.

**Comment by liamcain** [comment]
I see. It looks like NotePlan uses the same behavior as Fantastical: Clicking the month will go back to the current month. This was the original behavior for the plugin but I switched it since it was less discoverable. I do understand your concern about the button being more prominent than it needs to be. I'll keep this ticket open and give it more thought before I do any more UI updates.

Thanks for raising!

**Comment by cristianvasquez** [comment]
I always see the text 'Today' and I instinctively click it. 

Perhaps something in the middle, that: 
1. bring you back to the current month if you're looking at a different month,
2. open today's note, otherwise.

would work?

**Comment by liamcain** [comment]
I'm currently planning a slightly larger design update that should address this (+ #61) where you can quickly jump to other months/years. For the short term, I think I will revert back to the previous behavior: get rid of the "today" button and allow clicking the month name to return to the current month.

I'll convert this issue to a Discussion for now so that people can share there thoughts on this.

**Comment by liamcain** [comment]
## Open Questions

## 1. If I remove the "Today" button, is there still a demand for quickly navigating to Today's note from the Calendar view?

I'm inclined to think that there are already enough ways to open "Today's daily note" without the calendar view and that users will prefer a cleaner interface to more buttons. The confusion is really just about the word "Today" and one it's gone, it will likely be a nonissue. Let me know if this intuition is off.

## 2. How can the behavior "Click the current month to return there" be more intuitive?

My plan is to just make the UI more similar to NotePlan, where hovering over the month draws a border around it (showing that it's a button). This should help with the UX a bit.

**Reply by tgrosinger** [reply]
I think you are right that if you removed the "Today" I would stop trying to use it to go to today's daily note. But currently with it there I click it all the time!

Thinking of something like Google Calendar, it has a "Today" button, and clicking the current month in the mini calendar does not do anything. I will however note that Google generally has horrible UX. The FastMail calendar also has a today button, and clicking on the month brings up a quick switcher. Slightly different use case, but related.

![image](https://user-images.githubusercontent.com/597386/104213170-dff8a880-53ea-11eb-8f2d-9e774d53a717.png)

**Reply by liamcain** [reply]
Is that screenshot from FastMail? @tgrosinger. I was picturing something like the "Go to date" for Slated. 

This UX change won't make it in the next version of the calendar but I will try to get to this soon!

**Reply by tgrosinger** [reply]
Yeah, that's FastMail. Oh yeah, definitely a good example for what the "Move" window should look like in Slated. I'll put this there too. Thanks!

**Reply by cristianvasquez** [reply]
> 1. If I remove the "Today" button, is there still a demand for quickly navigating to Today's note from the Calendar view?

Note Today's note does not fill the templates. I personally go through the calendar as a walk-around https://forum.obsidian.md/t/use-new-templates-plugin-for-daily-notes/4756/3

**Reply by liamcain** [reply]
@cristianvasquez, I think you're missing the formatting string in the template. It should work if you put `{{date:MM dd, YYYYY}}`

**Reply by cristianvasquez** [reply]
I can confirm that with a formatting string it works :)

Perhaps the calendar should call 'daily note' for that one to ensure the same behavior

**Reply by liamcain** [reply]
the build in daily note functionality doesn't support days that aren't today. That's why the calendar plugin replicates the functionality rather than reuses it. Regardless, I think that's pretty off-topic for this discussion.

**Reply by eleanorkonik** [reply]
> ## 1. If I remove the "Today" button, is there still a demand for quickly navigating to Today's note from the Calendar view?
> I'm inclined to think that there are already enough ways to open "Today's daily note" without the calendar view and that users will prefer a cleaner interface to more buttons. The confusion is really just about the word "Today" and one it's gone, it will likely be a nonissue. Let me know if this intuition is off.

For me, I _never_ remember the left icon button, which is totally separate from the calendar itself visually and just gets mixed in to a bunch of other stuff -- the icon doesn't scream "today," to me, or even "calendar." The icon makes it look like a task list, which isn't at all what I use the calendar plugin / daily note plugin for. Without mousing over, I have no idea it's the "open today's note" button and I mostly don't interact with those icons because I have a hard time remembering what they're for, other than the shuffle dice and the graph icon. While I can click on the date on the calendar (which is always up on my right bar) to open today's note,I use css to make the calendar pretty small, and my screen is big, so "today" doesn't stand out very well -- I have to actively remember what today's date is, which, as a stay at home mom... I barely remember what month it is, if I'm being honest. Once I go back to work teaching I'll probably stop forgetting the date, tho :P 

I'm not really a hotkey user -- I'm a very mouse-first user, I only learn hotkeys if I _have_ to -- so being able to click on "today" to get today's note would be ideal now that I'm starting to use the daily notes for more things. 

> ## 2. How can the behavior "Click the current month to return there" be more intuitive?
> My plan is to just make the UI more similar to NotePlan, where hovering over the month draws a border around it (showing that it's a button). This should help with the UX a bit.

I did not know you could do this until I saw this comment, so you're right that it's not intuitive that "today" doesn't work but "MArch 2021" will take me back to the calendar that shows me today. If I were to describe what the most intuitive behavior for me would be, it would be that if I'm on APR 2021 and click "today" it snaps back to FEB 2021 (which is basically how it works in gcal), then if I click it again, it opens today's note (and if I cmd+click, it opens it in a new window).
